osquery/tables/system/windows/chassis_info.cpp

```diff
@@ -0,0 +1,69 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+#include <osquery/system.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/windows/wmi.h"
+
+namespace osquery {
+namespace tables {
+
+QueryData genChassisInfo(QueryContext& context) {
+  QueryData results;
+
+  WmiRequest wmiSystemReq("select * from Win32_SystemEnclosure");
+  const auto& wmiResults = wmiSystemReq.results();
+  if (wmiSystemReq.getStatus().ok()) {
+    if (!wmiResults.empty()) {
```

Comment:
Flip this and early return

Reply:
So it's been quite some time, and I've forgotten some of the context. Do we want:
-----
if (!results.empty())
   return warning
else
   generate table
-----
as opposed to the current order, right?

Reply:
Close. The early return pattern is:
```
if (results.empty())
  return warning
end

<normal stuff>
```

Thus the normal stuff isn't indented.

Reply:
That makes alot more sense when you put it that way, hah. I'll do that.